src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java

```diff
@@ -128,21 +132,28 @@ public ListBoxModel doFillFileIdItems(@AncestorInPath ItemGroup context, @Ancest
          * @param fileId  the id of the config file
          * @return a validation result / description
          */
-        public HttpResponse doCheckFileId(StaplerRequest req, @AncestorInPath Item context, @QueryParameter String fileId) {
+        public HttpResponse doCheckFileId(StaplerRequest req, @AncestorInPath Item context, @AncestorInPath AccessControlled ac, @QueryParameter String fileId) {
             // You should have permission to configure your project in order to check whether the selected file id is
             // allowed to you
             if (context != null) {
                 context.checkPermission(Item.CONFIGURE);
+            } else {
+                ac.checkPermission(Item.CREATE);
```

Comment:
As above, but return `FormValidation.OK` for bad input if permissions are missing (perhaps with a note that permissions to check are missing).